components/dashboard/src/App.tsx

```diff
@@ -168,27 +168,16 @@ function App() {
         (async () => {
             var user: User | undefined;
             try {
-                let teamsPromise: Promise<Team[]>;
-                if (usePublicApiTeamsService) {
-                    teamsPromise = teamsService.listTeams({}).then((response) => {
-                        return response.teams.map((team) => {
-                            const t: Team = {
-                                id: team.id,
-                                name: team.name,
-                                slug: team.slug,
-                                creationTime: "",
-                            };
-                            return t;
-                        });
-                    });
-                } else {
-                    teamsPromise = getGitpodService().server.getTeams();
-                }
-
                 user = await getGitpodService().server.getLoggedInUser();
                 setUser(user);
 
-                const teams = await teamsPromise;
+                let teams: Team[];
+                if (usePublicApiTeamsService) {
+                    const response = await teamsService.listTeams({});
+                    teams = mapPublicApiTeamsToProtocol(response.teams);
+                } else {
+                    teams = await getGitpodService().server.getTeams();
+                }
```

Comment:
Pushing the team fetching after the user does 2 things:
1. ensures that we have a user when we fetch teams - this makes the feature flag controllable by user
2. It cleans up the logic a bit